### ISSUE TITLE
Questioning the unreasonableness of the use of optical flow in this paper

### ISSUE THREAD

**Issue by MaxChanger** [GewelsJI/GTNet#3]
As mentioned by the author of this paper, the color optical flow image generated by RAFT and loaded in as a color Image (3-channel RGB) is used in the code.
https://github.com/GewelsJI/GTNet/blob/0c0c30dceff0d49d4d948b33079c57c934b24f95/dataloaders/davsod.py#L44

The value of flow image you used is necessarily in 0-1 or 0-255, while this color picture in the field of optical flow is only applicable to the visualization of optical flow, basically there is no practical significance, except that it can be interpreted as a way to intuitively see the difference between the motion of the foreground moving objects and the background, does not contain other motion information. In addition to this, the Motion information between images will be lost, and **there will be no** so-called 'Motion-guided Mask Propagation' in your paper. 

Why it is possible to use visualized colored optical flow as input here is very unreasonable in my opinion.

I hope the author can answer my questions in a reasonable way.

**Comment by MaxChanger** [comment]
Ping some some active people @GewelsJI @WangXiao213 
Hope to get your reply.

**Comment by GewelsJI** [comment]
Hi, @MaxChanger 

Thank you for your attention to our work. We follow the same protocols of optical flow generation suggested by [2019-ICCV-MGA](https://github.com/lhaof/Motion-Guided-Attention). Note that the optical map is generated from the flow generator, such as FlowNet2.0 and RAFT.

Indeed, the shape of the raw flow map from the generator is [H, W, 2]. For a better demonstration, we visualize it as the RGB format with shape [H, W, 3]. The RGB format also includes the motion information of the movements in the current scene due as it is the transformation of the raw flow map. We refer you to 2019-ICCV-MGA (Motion Guided Attention for Video Salient Object Detection) or 2020-TIP-MATNet (Motion-Attentive Transition Network for Zero-Shot Video Object Segmentation). Some ideas and effectiveness have been proved in those prior works. Thus, it provides the affluent motion cues in the propagation of neural network.

Besides, based on my previous limited attempts, the raw optical flow map with two channels is also working when it serves as the input of the motion branch. What my initial understanding is that the transformation from raw flow map to RGB flow map belongs to the rule-based process. Thus, this process can be automatically learnt from the neural network. If you have some progress in this attempt, I am pleased to talk with you for further discussion.

Best regards,
Ge-Peng.

**Comment by MaxChanger** [comment]
Thank you for the quick reply.👏 @GewelsJI 

I checked the two papers you mentioned and their code, and it is indeed like you said, the former also uses 3-channel RGB optical flow, and again you agree with their way, so you used it that way.

**But as I said above, the 3-channel RGB map contains more semantic information than motion information, and the pixel-level motion information will be lost after mapping the flow vector delta_x and delta_y to the 0-255 color space.** For a sample example, you can use a 2-channel raw optical flow map to warp or propagate the object mask, but a 3-channel RGB map does not have this capability, unless it is remapped back to flow sapce according to the color wheel. So I think what you said about  *The RGB format also includes the motion information* is unreasonable. (While it is possible to respond to me with the former having done so, I am more interested in hearing your understanding of this)

I'm glad to hear that you have experimented with a 2-channel raw optical flow map, based on your logic, the result is worse than using a 3-channel RGB optical flow map. Can you further explain the reasons in more detail, or give me some experimental results to verify your conclusions, if you can find them.

**Comment by GewelsJI** [comment]
Hi, @MaxChanger 

Q1: The understanding of using a 3-Channel RGB flow map.
First, I am not sure that this #idea (the 3-channel RGB map contains more semantic information than motion information) is supportable from my side. I conduct initial experiments, and it shows a slight difference between a 2-channel raw flow map and a 3-channel RGB flow map. And I did not give an in-depth investigation for it. Second, I treat the mapping process as the rule-based transformation. This process has similar spirits to the HHA transformation in the RGB-D field. (2014-ECCV-Learning rich features from RGB-D images for object detection and segmentation)

Q2: What is the essential reason why 3-channel RGB is better than 2-channel raw flow map?
I never say that the performance of the 2-Channel variant is better than the 3-Channel one. We follow the same protocol proposed in MGA and MATNet. This training strategy is also working and reasonable for the video object segmentation (VOS) and video salient object detection (VSOD) tasks. You can google some related papers using these keywords. 

Suggestion:
1. Is there any evidence or some verification for your #idea? Or it just is based on your intuitive or empirical experience? 
2. Can you verify your #idea via our code, MGA, MATNet? For example, you can conduct comparison experiments of 2-Channel and 3-Channel variants. And then, you can post the result here, and we can talk further.

Best regards,
Ge-Peng.

**Comment by MaxChanger** [comment]
For Q1, I gave a very simple example above, I hope you can understand it carefully.

For Q2, please note that I said *"you have experimented with a 2-channel raw optical flow map, based on your logic, the result is worse than using a 3-channel RGB optical flow map. "* and not *"2-Channel variant is better than the 3-Channel one"*. You may not have looked closely enough. Because I heard you say you did 2-channel experiments, so if 2-channel is better than 3-channel in your experimental setup, why not report better 2-channel results in the paper? Is it just to follow the previous setup, I don't think it should be so, so I conclude that in your experiment 2 channel is worse than 3 channel, am I right?

For Suggestion, I think I have some in-depth knowledge of optical flow and VOS, and the issue we are discussing is not related to the downstream task of salience, so please be at peace.

**Comment by GewelsJI** [comment]
I apologize for the temporary closure. I find some bugs in our `README.md` due to our cursoriness. We revise the misleading contents, and now you can enjoy it.

For Q1. Yes. You are right. After discussion with e-mail and GitHub-issue, I agree with the idea you claimed. In search of truth, some ideas require further verification via experiments.

For Q2. Very sorry for the above misleading sentence. Some clarification refer to this [reply](https://github.com/GewelsJI/GTNet/issues/3#issuecomment-853448286)

To this end, the idea claimed is the nice motivation that I have ever seen before in this field. This huge improvement may lead to the trend of whole field development. Thus, if you conduct the experiments I mentioned before, you can post the initial result here. I am pleased to talk and collaborate with such excellent authors further.

Thank you for your attention, and I hope you have a happy working day.

Best regards,
Ge-Peng.

**Comment by MaxChanger** [comment]
It was a pleasure to have an in-depth discussion with the excellent author of this article.

I think this issue can be closed for now, and others who are interested are welcome to continue to participate in this discussion.

Thanks again. @GewelsJI
